enhancements/rhcos/liveisoinstall.md

```diff
@@ -0,0 +1,93 @@
+---
+title: RHEL CoreOS Live ISO/installer
+authors:
+  - "@cgwalters"
+reviewers:
+  - "@coreos-team"
+approvers:
+  - "@coreos-team"
+creation-date: 2020-02-07
+last-updated: 2020-02-07
+status: implementable
+---
+
+# Rebase RHEL CoreOS 4.5 on Fedora CoreOS $latest
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+RHEL CoreOS is built from RHEL components, but uses configuration and tooling from Fedora CoreOS.  They share many things such as the installer code, etc.
+
+This enhancement tracks rebasing RHCOS 4.5 on the latest FCOS code, in particular the installer.
+
+### New "Live" installer
+
+Currently, RHCOS 4.4 and below offer a kernel/initramfs and ISO that can run a shell script installation program from the initramfs.
+
+In FCOS $latest, there is a Live ISO/PXE media that runs the full OS (for example, it includes `podman`), and a [coreos-installer](https://github.com/coreos/coreos-installer/) program was rewritten.
+
+## Motivation
+
+This allows administrators to run arbitrary code before and after the installation process by providing Ignition that wraps the installer.  For example, higher level tooling like Foreman and Ironic can easily `curl` a URL to report provisioning success, or even pull down and launch a container before/after.
+
+An additional goal is to allow administrators to more easily run RHCOS as a "live" system for *interactive* experiementation and testing.  For example, an administrator could run the system from a USB stick to validate the network interface names, etc.
```

Comment:
Do we know if this is a strong desire for customers?  Is this something we want to officially support?

It reminds me of RHELAH `devmode` which didn't seem to get a lot of traction with customers.

Reply:
I see no reason not to support at least the two use cases of "run an interactive shell" and "run coreos-installer".

There's a fuzzy line for sure, shipping this makes it easy for users to do whatever they want on  bare metal *without* installing, but in practice that's just a small generalization of supporting anything *after* installing.  (Except for the big one of joining a cluster)

As far as devmode - hmm, I see that more as just solving the "how do I log in" problem which we actually *still* have since we don't have `coreos.autologin`, not the "how do run code around installing" problem.

Reply:
> I see no reason not to support at least the two use cases of "run an interactive shell" and "run coreos-installer".

I guess I'm just worried about what kind of support requests will get generated by enabling this and how CEE will respond to that.

I see value in enabling both of those use cases but let's also recognize there may a support cost related to that.